src/couch_views/src/couch_views_http.erl

```diff
@@ -108,7 +108,7 @@ paginated_cb({meta, Meta}, #vacc{}=VAcc) ->
 paginated(Req, EtagTerm, #mrargs{page_size = PageSize} = Args, KeyFun, Fun) ->
     Etag = couch_httpd:make_etag(EtagTerm),
     chttpd:etag_respond(Req, Etag, fun() ->
-        hd(do_paginated(PageSize, [set_limit(Args)], KeyFun, Fun))
+        hd(do_paginated(PageSize, [Args], KeyFun, Fun))
```

Comment:
For curious reviewer we were setting limit here and then increment it again on line 127.